services/src/ogc/wms/request.rs

```diff
@@ -0,0 +1,199 @@
+use serde::{Deserialize, Serialize};
+use crate::util::from_str;
+use geoengine_datatypes::primitives::{BoundingBox2D, Coordinate2D, TimeInterval};
+use serde::de::Error;
+
+// TODO: ignore case for field names
+
+#[derive(PartialEq, Debug, Deserialize, Serialize)]
+#[serde(tag = "request")]
+#[allow(clippy::pub_enum_variant_names, clippy::large_enum_variant)]
+pub enum WMSRequest {
+    GetCapabilities(GetCapabilities),
+    GetMap(GetMap),
+    GetFeatureInfo(GetFeatureInfo),
+    GetStyles(GetStyles),
+    GetLegendGraphic(GetLegendGraphic),
+}
+
+#[derive(PartialEq, Debug, Deserialize, Serialize)]
+pub struct GetCapabilities {
+    pub version: Option<String>,
+    pub service: String,
+    pub format: Option<String>, // TODO: Option<GetCapabilitiesFormat>,
+}
+
+#[derive(PartialEq, Debug, Deserialize, Serialize)]
+pub enum GetCapabilitiesFormat {
+    TextXml //, ...
```

Comment:
Yes, XML is default and required. Thanks to _serde_ this is easier than before(?).